include/integratorxx/quadratures/gausslegendre.hpp

```diff
@@ -44,63 +70,51 @@ struct quadrature_traits<
   using weight_container = std::vector< weight_type >;
 
   inline static std::tuple<point_container,weight_container>
-    generate( size_t npts, point_type lo, point_type up ) {
-
-    assert( npts % 2 == 0 );
-    assert( lo != -std::numeric_limits<double>::infinity() );
-    assert( up != std::numeric_limits<double>::infinity() );
-
+  generate( size_t npts ) {
     point_container  points( npts );
     weight_container weights( npts );
 
+    // Absolute precision for the nodes
+    const double eps=3.0e-11;
+
+    // Since the rules are symmetric around the origin, we only need
+    // to compute one half of the points
     const size_t mid = (npts + 1) / 2;
-    const double eps = 3.e-11; // Convergence tolerance
-
-    for( size_t i = 1; i <= mid; ++i ) {
-
-      const point_type ip = i;
-
-      point_type z = std::cos( M_PI * (ip - 0.25) / (npts + 0.5));
-      point_type pp(0), z1(0);
-
-      // Iteratively determine the i-th root 
-      while(std::abs(z-z1) > eps) {
-        point_type p1(1.), p2(0.);
-  
-        // Loop over the recurrence relation to evaluate the
-        // Legendre polynomial at position z
-        for( size_t j = 1; j <= npts; ++j ){
-          const point_type jp = j;
-
-          point_type p3 = p2;
-          p2 = p1; 
-          p1 = ( (2. * jp - 1.) * z * p2 - (jp - 1.) * p3) / jp;
-        } // end j for
-  
-        // p1 is now the desired Legrendre polynomial. We next compute 
-        // pp, its derivative, by a standard relation involving also p2,
-        // the polynomial of one lower order
-        pp = npts * (z * p1 - p2) / (z * z - 1.);
-        z1 = z;
-        z  = z1 - p1 / pp;
-      } // end while
-  
-      point_type  pt = z;
-      weight_type wgt = 2. / (1. - z*z) / pp / pp;
-
-      // Transform points and populate arrays
-      std::tie(pt,wgt) = transform_minus_one_to_one(lo,up,pt,wgt);
-      points[i-1] = pt;
-      weights[i-1] = wgt;
-
-      // Reflect the points
-      points[npts-i]  = (lo + up) - pt;
-      weights[npts-i] = wgt;
-
-    }; // Loop over points
+    for( size_t idx = 0; idx < mid; ++idx ) {
+        // Index
+        const point_type i = idx+1;
+
+        // Standard initial guess for location of i:th root in [-1, 1]
+        point_type z = std::cos( M_PI * (i - 0.25) / (npts + 0.5));
+        // Old value of root
+        point_type z_old = -2.0;
+        // Values of P_n(x) and dP_n/dx
+        point_type p_n, dp_n;
+
+        // Solve the root to eps absolute precision
+        while(std::abs(z-z_old) > eps) {
+          // Evaluate the Legendre polynomial at z and its derivative
+          eval_Pn<point_type>(z,npts,p_n,dp_n);
```

Comment:
Is there a reason that you're manually specifying the template parameter here? It should be able to be deduced from `z`

Reply:
Removed